server/graphql/v2/mutation/AccountMutations.ts

```diff
@@ -0,0 +1,50 @@
+import { GraphQLNonNull } from 'graphql';
+import GraphQLJSON from 'graphql-type-json';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { Account } from '../interface/Account';
+import { Unauthorized, Forbidden } from '../../errors';
+import AccountSettingsKey from '../enum/AccountSettingsKey';
+import { sequelize } from '../../../models';
+
+const accountMutations = {
+  editAccountSetting: {
+    type: new GraphQLNonNull(Account),
+    description: 'Edit the settings for the given account',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account where the settings will be updated',
+      },
+      key: {
+        type: new GraphQLNonNull(AccountSettingsKey),
+        description: 'The key that you want to edit in settings',
+      },
+      value: {
+        type: new GraphQLNonNull(GraphQLJSON),
+        description: 'The value to set for this key',
+      },
+    },
+    async resolve(_, args, req): Promise<object> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      return sequelize.transaction(async transaction => {
+        const account = await fetchAccountWithReference(args.account, {
+          dbTransaction: transaction,
+          lock: true,
+          throwIfMissing: true,
+        });
+
+        if (!req.remoteUser.isAdmin(account.id)) {
+          throw new Forbidden();
+        }
+
+        const settings = { ...account.settings, [args.key]: args.value };
```

Comment:
We should use `lodash.set` to support nested properties.

Reply:
Interesting idea, I didn't think about that. It will make whitelisting and usage a bit less straightforward because we won't be able to use the `AccountSettingsKey` enum anymore for the `key` field but updating nested values will be safer.

@opencollective/core-developers any opinion on this?

Reply:
I made the update, the safety and usage are really better when we allow nested keys.

```es6
editAccountSetting({ account, key: 'collectivePage.background.zoom', value: 0.5 }) 
```

Thanks @kewitz for the suggestion :+1: